src/mam4xx/aero_model.hpp

```diff
@@ -0,0 +1,634 @@
+  !-----------------------------------------------------------------*/
+
+  // @param [in]  na           ! number of aerosol bins
+  // @param [in]  xlo          ! lower limit of aerosol radius (log)
+  // @param [in]  dx           ! aerosol radius bin width (log)
+  // @param [in]  xg0          ! log(mean radius)
+  // @param [in]  sx           ! standard deviation (log)
+
+  // @param [out] raerosv(:)   ! aerosol radius [cm]
+  // @param [out] fnumaerosv(:)! fraction of total number in the bin
+  // @param [out] fvolaerosv(:)! fraction of total volume
+  // in the bin [fraction]
+
+  // ! calculate total aerosol number and volume
+  const Real zero = 0;
+  // total aerosol number
+  Real anumsum = zero;
+  // total aerosol volume
+  Real avolsum = zero;
+  for (int ii = 0; ii < na; ++ii) {
+    const Real xx = xlo + ii * dx;
+    // aerosol radius in the bin [cm]
+    const Real aa = haero::exp(xx);
+    raerosv[ii] = aa;
+    const Real dum = (xx - xg0) / sx;
+    fnumaerosv[ii] = haero::exp(-0.5 * dum * dum);
+    // 1.3333 is simplified 4/3 for sphere volume calculation
+    fvolaerosv[ii] =
+        fnumaerosv[ii] * 4 * haero::Constants::pi * aa * aa * aa / 3;
+    anumsum += fnumaerosv[ii];
+    avolsum += fvolaerosv[ii];
+  } // end ii
+
+  // ! calculate fraction in each aerosol bin
+  for (int ii = 0; ii < na; ++ii) {
+    fnumaerosv[ii] /= anumsum;
+    fvolaerosv[ii] /= avolsum;
+  } // end ii
+
+} // calc_aer_conc_frac
+
+KOKKOS_INLINE_FUNCTION
+void calc_schmidt_number(const Real freepath, const Real r_aer,
+                         const Real temp, //& ! in
+                         const Real rhoaero, const Real rhoair,
+                         const Real airkinvisc,         // & ! in
+                         Real &schmidt, Real &taurelax) //! out
+{
+  /*-----------------------------------------------------------------
+  ! calculate Schmidt number
+  ! also output relaxation time for Stokes number
+  !
+  ! note that there is a similar calculation of Schmidt number in dry
+  ! depositon (in modal_aero_drydep.F90) but the calculation of dumfuchs (or
+  ! slip_correction_factor) looks differently
+  !-----------------------------------------------------------------*/
+
+  // @param [in]  freepath      ! molecular freepath [cm]
+  // @param [in]  r_aer         ! aerosol radius [cm]
+  // @param [in]  temp          ! temperature [K]
+  // @param [in]  rhoaero       ! density of aerosol particles[g/cm^3]
+  // @param [in]  rhoair        ! air mass density [g/cm^3]
+  // @param [in]  airkinvisc    ! air kinematic viscosity [cm2/s]
+
+  // @param [out]  schmidt       ! Schmidt number [unitless]
+  // @param [out]  taurelax      ! relaxation time for Stokes number
+  // [s]
+
+  // BAD CONSTANT
+  const Real boltz =
+      haero::Constants::boltzmann; // 1.38065e-23;      //  ! Boltzmann's
+                                   // constant ~ J/K/molecule
+  // BAD CONSTANT
```

Comment:
This looks to me like a straight conversion from SI (mks, Joules/K) to cgs (ergs/K) units, which explains the conversion factor. Do you agree?

Reply:
Yes, I do.